### DISCUSSION TITLE
[Question]: How should images be returned from actions so they are displayed?

### DISCUSSION THREAD

**Discussion by dividor** [danny-avila/LibreChat#2576] [Q&A]
### What is your question?

We have an action that can return images (graphs) as base64 encoded strings. Is there a way to display them directly rather than passing the action output through the LLM?

### More Details

Please see above

### What is the main subject of your question?

Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Tools will be easier to add, including defining what the frontend side of it looks like. I've already prototyped a solution and it works well!

But since actions are meant to be flexible by design, maybe there can be an additional setting for them. Will think about that too.
